packages/react-testing/src/root.tsx

```diff
@@ -140,10 +136,38 @@ export class Root<Props> implements Node<Props> {
           return result as unknown as Promise<void>;
         }
 
-        return Promise.race([
-          result,
-          this.actPromise,
-        ]) as unknown as Promise<void>;
+        /* return a thenable which when invoked will add react's callback (to clear queue)
+         * into a set. If the result ever resolves we will remove the callback from the Set.
+         * If it doesn't we will call all callbacks in the Set when this root is destroyed which
+         * will unblock any stuck queues allowing subsequent tests to run
+         * Note: This can be cleanly achieved with a `Promise.race` but that adds an extra micro-task
```

Comment:
Added a comment about `Promise.race` in case anyone would be tempted to refactor this in the future 😓

Reply:
Ah good call, I'll see if I can put a test case together later. It's going to quite tricky to extract a minimal test case from that `apollo-client` mess